utils/s3/creds.hh

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <seastar/core/shared_ptr.hh>
+
+namespace s3 {
+
+struct endpoint_config {
+};
+
+using endpoint_config_ptr = seastar::lw_shared_ptr<endpoint_config>;
+
+} // s3 namespace
```

Comment:
I didn't understand the commend about the trust store configuration. Maybe we should extend the dbuild environment with the certificates? I don't see how we could have created a container without them.

Or is this about the relocatable package thunk? We already do [something](https://github.com/scylladb/scylladb/blob/c501163f955e8f665211fb1456f92fcedf07dd3d/install.sh#L163), maybe we need to add more?

Reply:
Here's what happens.

The call to `set_system_trust` translates into `gnutls_certificate_set_x509_system_trust()`. When I build scylla or seastar app with `ninja <mode>-build` and run the binary the latter gnutls call eventually does `open(/etc/ssl/certs/ca-certificates.crt)`, apparently loads root cerificates from it, and subsequent `tls_socket::connect()` works.

When I build relocatable package with `ninja dist-server-deb` and install and run the resulting binary the call to above gnutls function doesn't result in opening _any_ file, so the creds are left in a state without root certs at all and subsequent `tls_socket::connect()` fails with the `verification_error` being unable to validate AWS certificates.

Reading ... well -- glancing through -- the gnutls sources I could find any non-compile-time reason for that, which made me thing that it's the relocatable package's version of gnutls library to blame.

Reply:
```console
$ rpm -q gnutls; ./tools/toolchain/dbuild rpm -q gnutls
gnutls-3.8.0-2.fc37.x86_64
gnutls-3.8.0-2.fc37.x86_64
```

Reply:
Or we could supply it programatically, reading the environment variable ourselves, since I don't see it in gnutls documentation.

Reply:
Whatever the path is, gnutls should try to access this file somehow, be it open()/stat()/access()/whatever. I saw no signs of it in strace output, but I'll re-check. And will try feeding it SSL_CERT_FILE explicitly